### ISSUE TITLE
Feature - directive to create TOC

### ISSUE THREAD

**Issue by melck** [marp-team/marp-core#136]
I would like to have a directive to show a TOC with page numbers and link to slide. 

Ideally with options to customize level of toc or ignore some title.

What do you think about that ?

**Comment by yhatt** [comment]
Marp Core is hesitant adding any more features without consensus by community. Marp's built-in feature must be based on [CommonMark](https://commonmark.org/), and we think that built-in features to enhance Markdown would have to be well-known syntax in Markdown and around communities.

TOC is available in [over 15 markdown-it plugins](https://www.npmjs.com/search?q=keywords%3Amarkdown-it-plugin%20toc), and [Marp (Marpit) parser can extend by plugin](https://github.com/marp-team/marp-cli#functional-engine). You can try to add TOC feature through available plugin by yourself (or can create another TOC plugin optimized for Marp).

```javascript
// An example of engine.js for Marp CLI's custom engine
const { Marp } = require('@marp-team/marp-core')
const markdownItTableOfContents = require('markdown-it-table-of-contents')

module.exports = opts => new Marp(opts).use(markdownItTableOfContents)
```

We may consider to add if our community showed TOC plugin is worth to add as built-in feature of Marp.

**Comment by melck** [comment]
Yes atm i have setup TOC like this, but i think it could be worth to add it as built-in plugin.

**Comment by yhatt** [comment]
I don't think that. Our core must keep simple and the least features to be continued maintenance. We don't want burning out again [by much requests just like an old Marp](https://github.com/yhatt/marp/issues?q=is:issue%20is:closed). It's a reason why adopted pluggable architecture.

Marp would have taken TOC plugin feature as built-in if it was well-known and taken in a lot of Markdown communities, but actually it does not. We are thinking users who want TOC are in the minority so it should provide as a third-party plugin.

**Comment by melck** [comment]
I understand better. This is mainly to have support integrated into vscode. I will wait for the external engine support by the vscode plugin.

**Comment by cmi1993** [comment]
If I  use Marp in vscode ,if there has any vscode plugins that can offer TOC feature？any Recommend？

**Comment by Chiogros** [comment]
A tip I do to get a clickable TOC: use links to page's anchors.
```md
<!-- _class: toc -->

# Table of content

1. [What is Marp](#what-is-marp)
2. [Why this theme](#why-this-theme)
3. [Available features](#available-features)

---

# What is Marp

---

# Why this theme

---

# Available features
